Code/GraphMol/Wrap/Mol.cpp

```diff
@@ -199,6 +200,101 @@ namespace RDKit {
     return mol.getNumAtoms(onlyExplicit);
   }
 
+  //----------------------------------------------------------------------------
+  // Atom Bookmark Interface
+  void SetAtomBookmark(ROMol *mol, Atom *atom, int mark)
+  {
+    // dangerous as there is not concept of shared pointers here (or internal
```

Comment:
I agree that the lifetime thing here is a real concern. If the Atom you are working with on the Python side goes out of scope, the bookmark becomes a liability.

Isn't this solvable with some of the boost.python functionality for indicating pointer ownership?